rpc/nodeagent/compute/virtualmachine/moc_nodeagent_virtualmachine.proto

```diff
@@ -108,6 +108,16 @@ message SecurityConfiguration {
 	SecurityType securityType = 3;
 }
 
+message ZoneReference {
+	string name = 1;
+	repeated string nodes = 2;
+}
+
+message ZoneConfiguration {
```

Comment:
Looks like there are duplicates in `ZoneConfiguration` declarations with minor differences. Is it possible to merge them?

Reply:
From CLI to cloud agent only zone names are passed. 
Since zones to node mapping  is only on cloud agent side,  when sending Create VM request to node agent , I am including node names associated with zone. 
Hence the minor difference in spec.

Merging them will lead to an unused field (nodes) on cloud agent side. Since it will be reused by placement group spec, Piyush suggested to keep them separate to keep it clean. 

Please let me know if merging is preferred.

Reply:
We can keep them separate in this case. Can we add this explanation in the comments so that others are not confused in the future?

Reply:
Add comments for defining separately for cloud agent and node agent.

Zone configuration on cloud agent will be used by VM and Placement Group, so moved it to common based on feedback from Piyush.